src/plugins/workspace/public/plugin.ts

```diff
@@ -194,7 +199,9 @@ export class WorkspacesPlugin implements Plugin<{}, {}, WorkspacesPluginSetupDep
     if (this.coreStart) {
       return this.coreStart.workspaces.client.currentWorkspaceId$.subscribe(
         (currentWorkspaceId) => {
-          this.coreStart?.savedObjects.client.setCurrentWorkspace(currentWorkspaceId);
+          this.coreStart?.savedObjects.client.setCurrentWorkspace(
+            currentWorkspaceId || PUBLIC_WORKSPACE
```

Comment:
I feel url should be the single source of truth, if need to enter to `public` workspace, it should navigate to /w/public`

Reply:
Yes, after discussion on some special types of workspaces, we should remove this default logic here.